### ISSUE TITLE
[YCQL] YCQL select max, min queries returning 2 rows for each.

### ISSUE THREAD

**Issue by Arjun-yb** [yugabyte/yugabyte-db#13295]
Jira Link: [DB-2929](https://yugabyte.atlassian.net/browse/DB-2929)
### Description

Version: **2.14.0.0-b75**

Steps:
1. Create universe with below GFLAGs
```
Master:
{"enable_automatic_tablet_splitting": "true", 
"tablet_split_high_phase_shard_count_per_node": 10000, 
"tablet_split_high_phase_size_threshold_bytes": 10485760, 
"tablet_split_low_phase_size_threshold_bytes": 1048576, 
"tablet_split_low_phase_shard_count_per_node": 16,
"db_write_buffer_size":102400}

Tserver:
{"enable_automatic_tablet_splitting":"true",
"yb_num_shards_per_tserver":1,
"ysql_num_shards_per_tserver":1,
"db_write_buffer_size":102400}
```
2. Create table and index
```
create table employees(id int primary key, name text, age int, description text) with transactions = { 'enabled' : true };
create index idx_2 on employees(name);
```
3. Load 100K rows to the table(with normal insert queries)
4. Load 600K more rows (with COPY cmd)
5. Run below queries and observe the output
```
select min(id), max(id) from employees;
select min(id) from employees;
select max(id) from employees;
```

**Actual**: 
Getting 2 rows(results) for each queries
```
ycqlsh:demo2> select min(id), max(id) from employees;

 system.min(id) | system.max(id)
----------------+----------------
              1 |         700000

---MORE---
 system.min(id) | system.max(id)
----------------+----------------
              3 |         699998

(2 rows)
ycqlsh:demo2> select min(id) from employees;

 system.min(id)
----------------
              1

---MORE---
 system.min(id)
----------------
              3

(2 rows)
ycqlsh:demo2> select max(id) from employees;

 system.max(id)
----------------
         700000

---MORE---
 system.max(id)
----------------
         699998

(2 rows)
```

**Comment by pkj415** [comment]
Did some debugging -- this seems like a day 0 issue due to our paging logic for aggregates and is unrelated to tablet splitting. Also, we don't need 600k rows to repro this, just 2 rows are enough if we pick a page size of 1.

The issue is that when we calculate aggregates in YCQL, we do so for each of the pages that we read. Quite weird that we never caught this issue. I say day 0 because this is an issue in 2.12 as well as seen below
```
Piyushs-MacBook-Pro:yugabyte-db-2.12 piyushjain$ ./bin/ycqlsh
Connected to local cluster at 127.0.0.1:9042.
[ycqlsh 5.0.1 | Cassandra 3.9-SNAPSHOT | CQL spec 3.4.2 | Native protocol v4]
Use HELP for help.
ycqlsh> create keyspace temp;
ycqlsh> use temp;
ycqlsh:temp> create table test (k int primary key, v int) with transactions = { 'enabled' : true };
ycqlsh:temp> insert into test (k, v) values (1, 1);
ycqlsh:temp> insert into test (k, v) values (2, 2);
ycqlsh:temp> paging 1;
ycqlsh:temp> select min(k), max(k) from test;

 system.min(k) | system.max(k)
---------------+---------------
             1 |             2

(1 rows)
ycqlsh:temp>
ycqlsh:temp> paging 1;
Page size: 1
ycqlsh:temp> select min(k), max(k) from test;

 system.min(k) | system.max(k)
---------------+---------------
             1 |             1

---MORE---
 system.min(k) | system.max(k)
---------------+---------------
             2 |             2

(2 rows)
ycqlsh:temp>
```

Will update here once I have more details

**Comment by OlegLoginov** [comment]
Reproduced with `paging 1` only:
```
ycqlsh:k> paging 2;
Page size: 2

ycqlsh:k> select min(id) from employees;
 system.min(id)
----------------
              1
(1 rows)

ycqlsh:k> paging 1;
Page size: 1

ycqlsh:k> select min(id) from employees;
 system.min(id)
----------------
              1

---MORE--- 
 system.min(id)
----------------
              2
(2 rows)
```

**Comment by OlegLoginov** [comment]
```
ycqlsh:k> create table test (k int primary key, v int) with transactions = { 'enabled' : true };
ycqlsh:k> insert into test (k,v) values(1,1);

ycqlsh:k> paging 1;
Page size: 1
ycqlsh:k> select min(k) from test;
 system.min(k)
---------------
             1
---MORE---
 system.min(k)
---------------
          null
(2 rows)

ycqlsh:k> paging 2;
Page size: 2
ycqlsh:k> select min(k) from test;
 system.min(k)
---------------
             1
(1 rows)
```

**Comment by OlegLoginov** [comment]
```
ycqlsh:k2> create table employees(id int primary key, name text, age int, description text) WITH tablets = 10 AND transactions = { 'enabled' : true };
ycqlsh:k2> insert into employees (id,name,age,description) values (1,'n1',1,'d1');
ycqlsh:k2> insert into employees (id,name,age,description) values (2,'n1',1,'d1');
ycqlsh:k2> insert into employees (id,name,age,description) values (3,'n1',1,'d1');
ycqlsh:k2> insert into employees (id,name,age,description) values (4,'n1',1,'d1');
ycqlsh:k2> paging 2;
Page size: 2
ycqlsh:k2> select min(id),max(id),count(*) from employees;
 system.min(id) | system.max(id) | count
----------------+----------------+-------
              1 |              4 |     2
---MORE---
 system.min(id) | system.max(id) | count
----------------+----------------+-------
              2 |              3 |     2
(2 rows)
```

**Comment by OlegLoginov** [comment]
NOTE: The problem happens if the internal number of fetched rows reaches the PageSize (by default it's 5000).
(Important: total number of rows, number of tablets, how the rows are distributed between tablets, current `Paging` value = PageSize.)
Indexes and tablet splitting does not affect the issue.

**Comment by OlegLoginov** [comment]
Fixed in the Master, 2.14, 2.16.
